src/vivarium_public_health/population/data_transformations.py

```diff
@@ -165,6 +163,21 @@ def rescale_binned_proportions(
 
         pop_data.loc[max_bin.index, "age_end"] = age_end
         pop_data.loc[padding_bin.index, "age_start"] = age_end
+        # Update age to be within bounds
```

Comment:
I don't understand what this logic is doing (or really, what any of the logic in this function is doing), but the returns seem weird to me. Let's talk.

Reply:
At the very least it doesn't seem to be updating both sexes the same:

<img width="232" alt="image" src="https://github.com/user-attachments/assets/2d990143-2cb1-4603-951b-c3513a6063fb" />